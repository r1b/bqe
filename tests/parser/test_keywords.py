import pytest


@pytest.mark.parametrize(
    "sql",
    (
        pytest.param("SELECT date abc", id="select_kw_date_column"),
        pytest.param("SELECT abc date", id="select_kw_date_alias"),
        pytest.param("SELECT date date", id="select_kw_date_double"),
        pytest.param("SELECT format abc", id="select_kw_format_column"),
        pytest.param("SELECT abc format", id="select_kw_format_alias"),
        pytest.param("SELECT percent abc", id="select_kw_percent_column"),
        pytest.param("SELECT abc percent", id="select_kw_percent_alias"),
    ),
)
def test_nonreserved_keyword_ok(sql, assert_parse_tree):
    assert_parse_tree(sql)
