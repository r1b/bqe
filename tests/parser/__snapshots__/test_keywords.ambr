# serializer version: 1
# name: test_nonreserved_keyword_ok[select_kw_date_alias]
  '''
  SELECT abc date
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	abc
          as_alias
            ident	date
      None
  '''
# ---
# name: test_nonreserved_keyword_ok[select_kw_date_column]
  '''
  SELECT date abc
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	date
          as_alias
            ident	abc
      None
  '''
# ---
# name: test_nonreserved_keyword_ok[select_kw_date_double]
  '''
  SELECT date date
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	date
          as_alias
            ident	date
      None
  '''
# ---
# name: test_nonreserved_keyword_ok[select_kw_format_alias]
  '''
  SELECT abc format
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	abc
          as_alias
            ident	format
      None
  '''
# ---
# name: test_nonreserved_keyword_ok[select_kw_format_column]
  '''
  SELECT format abc
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	format
          as_alias
            ident	abc
      None
  '''
# ---
# name: test_nonreserved_keyword_ok[select_kw_percent_alias]
  '''
  SELECT abc percent
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	abc
          as_alias
            ident	percent
      None
  '''
# ---
# name: test_nonreserved_keyword_ok[select_kw_percent_column]
  '''
  SELECT percent abc
  
  query_expr
    select
      None
      None
      select_list
        select_expr
          ident	percent
          as_alias
            ident	abc
      None
  '''
# ---
